crates/pep508-rs/src/unnamed.rs

```diff
@@ -356,8 +362,31 @@ fn parse_unnamed_url<Url: UnnamedRequirementUrl>(
 ) -> Result<(Url, Vec<ExtraName>), Pep508Error<Url>> {
     // wsp*
     cursor.eat_whitespace();
+
     // <URI_reference>
-    let (start, len) = cursor.take_while(|char| !char.is_whitespace());
+    let (start, len) = {
+        let start = cursor.pos();
+        let mut len = 0;
+        let mut backslash = false;
+        let mut depth = 0u32;
+        while let Some((_, c)) = cursor.next() {
+            if depth == 0 && c.is_whitespace() {
+                break;
+            }
+            if backslash {
+                backslash = false;
+            } else if c == '\\' {
+                backslash = true;
+            } else if c == '[' {
+                depth += 1;
+            } else if c == ']' {
```

Comment:
Is it guaranteed that the parentheses are always balanced?

Reply:
Good call, thanks.